### ISSUE TITLE
Features want to see

### ISSUE THREAD

**Issue by greabock** [cycle/orm#151]
## Joined Table Inheritance (Class Table Inheritance)
It's like STI, but inheritor-specific fields are stored in a separate table.
related issue: https://github.com/cycle/orm/issues/116

## Polymorphic relations for inherited entities.
There is such a problem: the actual role of the parent entity and the child entity is same.
When we do a polymorphic relationship, an abstract role is specified in the table, and what this is concrete is not clear until the promise is resolved.
I think it would be possible to introduce a different, `role` for all descendants. I think this promises a lot of rework, but it seems to me that this would be the right decision. This, by the way, would eliminate the `discriminator` column.
Part of the problem with inheritance, by the way, is described here https://github.com/cycle/orm/issues/97. But it seems to me that the discussion here went in the wrong direction and we need different solution

## Always return a proxy object instead of an entity.
If not immediately out of the box, then at least an API that allows you to configure this behavior, and instructions to do it.
The problem I want to solve is very similar to https://github.com/cycle/orm/issues/132

## Search in identity map is performed only by PK.
I believe that it is possible to define some heuristics that will allow you to search also by unique fields, or unique groups of fields.
For example: a field is marked in the schema as unique, which means that you can also search for it.

**Comment by guilhermeaiolfi** [comment]
Polymorphic relations as you described seems the biggest missing feature of cycle. I don't know about you, but it's pretty common for my projects to have those relationships. I played around trying to implement it using custom mappers and went pretty far but gave up because of the deadline I had. But definitely something I would like to see in the core.

Something I would like to see also would be composite primary keys. But if I had to choose one, polymorphic relationships would be it.
